source/extensions/tracers/xray/tracer.cc

```diff
@@ -44,17 +140,59 @@ Tracing::SpanPtr Tracer::startSpan(const std::string& span_name, const std::stri
       span_ptr->setSampled(true);
       break;
     case SamplingDecision::NotSampled:
-      span_ptr->setSampled(false);
+      // should never get here. If the header has Sampled=0 then we never call startSpan().
+      ASSERT(false);
```

Comment:
doesnt hurt to `RELEASE_ASSERT()` in these kinds of cases.

Reply:
that's big guns!!